riscv/cfg.h

```diff
@@ -29,23 +29,25 @@ class cfg_t
 {
 public:
   cfg_t(std::pair<reg_t, reg_t> default_initrd_bounds,
-        const char *default_bootargs, size_t default_nprocs,
```

Comment:
Commit 72c119f03a148fc3512e85547deb531004f8f6e1 is awfully large. Things like moving the nprocs/hartid sanity checking, removing nprocs in lieu of hartids list, could/should be separate commits.

Reply:
Hmm, it's only +59/-41 (and the version I'm about to push up is +57/-37).

I'm not entirely sure how to split things up: as long as we're passing nprocs and hartids separately to `sim.cc`, I think the sanity check belongs there. But once we've merged the two fields, we (obviously) have to do the sanity check where we do the merge (in `spike.cc`). So I think that part of the commit is reasonably atomic. We *could* split up the "merge nprocs and hartids" and "move the result into cfg_t" change, but I'm not sure how much that buys: the second commit will just be a couple of lines.

Have I missed something?

Reply:
I don't think each commit needs to make complete sense in isolation. Moving the sanity check first might not make sense in isolation, but if it splits up a large commit, it can help with pinpointing future-discovered regressions. Even commits that are only a couple of lines are useful to separate.

Reply:
I wasn't trying hard enough :-) I've just done a bit of surgery and have unpicked things: that commit is now in 4 parts!